src/aspire/basis/pswf_2d.py

```diff
@@ -82,22 +82,11 @@ def _build(self):
     def _generate_grid(self):
         """
         Generate the 2D sampling grid
-
-        TODO: need to re-implement to use the similar grid function as FB methods.
         """
-        if self.nres % 2 == 0:
-            x_1d_grid = range(-self.rcut, self.rcut)
-        else:
-            x_1d_grid = range(-self.rcut, self.rcut + 1)
-        x_2d_grid, y_2d_grid = np.meshgrid(x_1d_grid, x_1d_grid)
-        r_2d_grid = np.sqrt(np.square(x_2d_grid) + np.square(y_2d_grid))
-        points_in_disk = r_2d_grid <= self.rcut
-        x = y_2d_grid[points_in_disk]
-        y = x_2d_grid[points_in_disk]
-        self._r_disk = np.sqrt(np.square(x) + np.square(y)) / self.rcut
-        self._theta_disk = np.angle(x + 1j * y)
-        self._image_height = len(x_1d_grid)
-        self._disk_mask = points_in_disk
+        grid = grid_2d(self.nres, normalized=False, indexing="xy")
```

Comment:
The xy-indexing is setting off alarm bells for me. I imagine this reproduces the original grid though. Should we be using yx everywhere?

Reply:
<img width="992" alt="Screenshot 2023-10-30 at 12 40 36 PM" src="https://github.com/ComputationalCryoEM/ASPIRE-Python/assets/47759732/b0a998b0-6ee3-4a18-9c90-323079a5dfe4">

Reply:
hey, at least this got rid of lines like

```
        x = y_2d_grid[points_in_disk]
        y = x_2d_grid[points_in_disk]
```  

🤣

Reply:
> hey, at least this got rid of lines like
> 
> ```
>         x = y_2d_grid[points_in_disk]
>         y = x_2d_grid[points_in_disk]
> ```
> 
> 🤣

An marked improvement for sure!